convert2rhel/actions/system_checks/check_firewalld_availability.py

```diff
@@ -0,0 +1,113 @@
+    if not os.path.exists(FIREWALLD_CONFIG_FILE):
+        return False
+
+    contents = ""
+    with open(FIREWALLD_CONFIG_FILE, mode="r") as handler:
+        contents = [line.strip() for line in handler.readlines() if line.strip()]
+
+    # Contents list is empty for some reason, better to assume that there
+    # is no content in the file that was read.
+    if not contents:
+        return False
+
+    # If the config file has this option set to true/yes, then we need to
+    # return True to ask the user to change it to self.
+    if list(filter(CLEANUP_MODULES_ON_EXIT_REGEX.match, contents)):
+        logger.debug("Found CleanupModulesOnExit option in firewalld.conf")
+        return True
+
+    logger.debug(
+        "Couldn't find CleanupModulesOnExit in firewalld.conf. Treating it as enabled because of default behavior."
+    )
+    # Return true anyway at the end even if the option was not found as the
+    # default value for CleanupModulesOnExit in the firewalld.conf is to be
+    # enabled always.
+    return True
+
+
+class CheckFirewalldAvailability(actions.Action):
+    id = "CHECK_FIREWALLD_AVAILABILITY"
+
+    def run(self):
+        """Error out if the firewalld service is running on the system."""
+        super(CheckFirewalldAvailability, self).run()
+        logger.task("Prepare: Check that firewalld is running")
+
+        if system_info.id == "oracle" and (system_info.version.major == 8 and system_info.version.minor >= 8):
+            if systeminfo.is_systemd_managed_service_running("firewalld"):
+                if _is_modules_cleanup_enabled():
+                    self.set_result(
+                        level="ERROR",
+                        id="FIREWALLD_MODULES_CLEANUP_ON_EXIT_CONFIG",
+                        title="Firewalld is set to cleanup modules after exit.",
+                        description="Firewalld running on Oracle Linux 8 can lead to a conversion failure.",
+                        diagnosis="We've detected that firewalld unit is running and that causes iptables and nftables failures on Oracle Linux 8 and under certain conditions it can lead to a conversion failure.",
+                        remediation=(
+                            "Set the option CleanupModulesOnExit in /etc/firewalld/firewalld.conf to no prior to running convert2rhel: "
+                            "sed -i -- 's/CleanupModulesOnExit=yes/CleanupModulesOnExit=no/g' /etc/firewalld/firewalld.conf\n"
+                            "You can change the option back to yes after the system reboot - that is after the system boots into the RHEL kernel."
+                        ),
+                    )
+                else:
+                    # Firewalld is running but the configuration for CleanupModulesOnExit was not set to true/yes
+                    description = (
+                        "We've detected that firewalld is running and we couldn't find check for the CleanupModulesOnExit configuration. "
```

Comment:
In my understanding, when `_is_modules_cleanup_enabled` returns False it should means that the CleanupModulesOnExit option is set to false.